rustual-boy-core/src/rom.rs

```diff
@@ -22,16 +22,22 @@ impl Rom {
         let mut vec = Vec::new();
         file.read_to_end(&mut vec)?;
 
-        let size = vec.len();
+        Rom::from_bytes(&vec)
+    }
+
+    pub fn from_bytes(bytes: &[u8]) -> io::Result<Rom> {
```

Comment:
I tend to advocate for taking the owned version of a type, if that's what you ultimately need (`Vec<u8>` in this case).

That way, in case the caller happens to have ownership of a Vec (like we do when loading from a file), the caller can just pass that, instead of needing an extra copy.